src/modules/webRTC/components/MultiConnectionChat.tsx

```diff
@@ -0,0 +1,118 @@
+import { ChatBox } from "./ChatBox";
+import { Offerings } from "./Offerrings";
+import { webRTC_connection } from "../baseWebRTC";
+import { ChatUserPage, UserInfo } from "./ChatUserPage";
+import { SidebarButton, SidebarPages } from "./SidebarButton";
+import { ModalWindow } from "../../components/ModalWindow";
+import { ChannelButton } from "./ChannelButton";
+import { chatTheme } from "./styles";
+
+export const WebRTCChat = () => {
+    const [page, setPage] = useState<SidebarPages>("chat");
+    const [userInfo, setUserInfo] = useState<UserInfo>({ name: "John Doe" });
+    const [actualPeerConnection, setActualPeerConnection] = useState<webRTC_connection>();
+    const [peerConnections, setPeerConnections] = useState<webRTC_connection[]>([]);
+    const [selectedConnections, setSelectedConnections] = useState<webRTC_connection[]>([]);
+    const [pressedButtons, setPressedButtons] = useState<boolean[]>([true]);
+
+    useEffect(() => {
+        if (pressedButtons[0]) {
+            setSelectedConnections(peerConnections);
+            return;
+        }
+        const mappedConnections: webRTC_connection[] = []; 
+        pressedButtons.forEach((checked, index) => {
+            checked && mappedConnections.push(peerConnections[index - 1]);
+        });
+
+        setSelectedConnections(mappedConnections)
+    }, [pressedButtons, peerConnections])
+
+    const addNewConnection = useCallback(() => {
+        const newPeerConnection = new webRTC_connection();
+        setActualPeerConnection(newPeerConnection);
+        setPeerConnections([...peerConnections, newPeerConnection]);
+        setPressedButtons([...pressedButtons, true])
+    }, [setPeerConnections, setActualPeerConnection, peerConnections]);
+
+    const newOfferClick = useCallback(() => {
+        addNewConnection();
+        setPage("offerings");
+    }, [addNewConnection, setPage]);
+
+    const newAnswerClick = useCallback(() => {
+        addNewConnection();
+        setPage("answerings");
+    }, [addNewConnection, setPage]);
+
+    const channelButtonClick = useCallback((index: number) => {
+        const newButtonsArray = [...pressedButtons];
+        newButtonsArray[index] = !pressedButtons[index];
+        setPressedButtons(newButtonsArray);
+    }, [pressedButtons, setPressedButtons])
+
+    return (
+        <Box
+            position="fixed" 
+            top={0} 
+            right={0}
+            flexDirection="row"
+            width={330}
+            height="100%"
+            minHeight={400}
+            gap={12}   
+            id="chat-root"         
+        >
+            <Box
+                id="channel-selection-container"
+                width={30}
+                padding={2}
+                flexDirection="column"
+            >
+                <ChannelButton key={"C-0"} active={pressedButtons[0]} text="All" onClick={() => channelButtonClick(0)}/>
+                {peerConnections.map((_, i) => {
+                    const index = i + 1;
```

Comment:
`i` and `index` at the same time is very confusing, handling 'All' in a different way would be very welcome